script/hassfest/mypy_config.py

```diff
@@ -0,0 +1,408 @@
+"""Generate mypy config."""
+from __future__ import annotations
+
+import configparser
+import io
+from typing import Final
+
+from .model import Config, Integration
+
+# Strictly typed parts of HA.
+# If your component is fully typed, please add it here to enable strict checks.
```

Comment:
Another approach would be to add an item in the manifest.json that would populate the corresponding list. Similar to how codeowners is generated.

Then we'd just have the ignore list here. New integrations should never end up on the ignore list.

Reply:
We can do this for components and add new field like `strictly_typed: true` in `manifest.json` but core modules don't have `manifest.json` so part of this list will still be needed.

But I think we can enable strict checks by default and then disable them for `components.*` and then enable them again per component basis. This way new modules in the core won't need to be added here and large part of this list can be removed.

Reply:
I think we should not add the typing_ignored item to the manifest since we should be working to remove that case completely.

> But I think we can enable strict checks by default and then disable them for components.* and then enable them again per component basis where needed.

Do you mean that we should do this and add the `strictly_typed` item to the manifest?

Reply:
Yes:
1. Enable strict check by default, this will be applied to all core modules.
2. Disable strict checks for `components.*`
3. Generate list of strictly typed components based on `strictly_typed` in their `manifest.json`.
4. Generate list of ingored components based on list in this hassfest plugin.

Reply:
Sounds good! 👍

Reply:
> I think we should not add the typing_ignored item to the manifest since we should be working to remove that case completely.

My idea is that can create more incentive for component developers to fix these errors quicker if they see it in their `manifest.json`.